src/binarySearchTree.js

```diff
@@ -0,0 +1,95 @@
+import BinaryNode from './binarynode'
+
+export default class BinarySearchTree {
+  constructor() {
+    this.length = 0
+    this.root = null
+  }
+
+  insert = (value) => {
+    const node = new BinaryNode(value)
+    if (this.count() == 0) {
+      this.root = node
+      this.length += 1
+    }
+    else {
+      let currentNode = this.root
+      for(let i = 0; i < this.count(); i++){
+        if(currentNode._value < node._value){
+          if(currentNode._rightChild == null){
+            currentNode._rightChild = node
+            this.length += 1
+            return
+          }
+          currentNode = this[i]._rightChild
+        }
+        if(currentNode._value > node._value){
+          if(currentNode._leftChild == null){
+            currentNode._leftChild = node
+            this.length += 1
+            return
+          }
+          currentNode = this[i]._leftChild
+        }
+      }
+    }
+  }
+
+  search = (value) => {
+    if(this.count() === 0){
+      return null
+    }
+    let currentNode = this.root
+    for(let i = 0; i < this.count(); i++){
+      if(currentNode._value == value){
```

Comment:
We should always use threequals (`===`) in order to avoid possible type coercion bugs.